spinta/types/model.py

```diff
@@ -190,6 +191,45 @@ def _parse_dtype_string(dtype: str) -> Tuple[str, List[str]]:
         return dtype, []
 
 
+def _check_denorm_prop(
+    name: str,
+    prop: Property,
+    model: Model,
+    manifest: Manifest,
+):
+    models = manifest.models
+    name_parts = name.split('.', 1)
+    name = name_parts[0]
+    properties = model.properties
+    if len(name_parts) > 1:
+        if name not in properties or not isinstance(properties[name].dtype, (Ref, Object)):
+            if prop.model == model:
+                raise NoRefPropertyForDenormProperty(
+                    ref=name,
+                    prop=prop.name,
+                    model=prop.model.name,
+                )
+            else:
+                raise UndefinedPropertyInDenormProperty(
+                    denorm_prop=prop.name,
+                    denorm_model=prop.model.name,
+                    ref_prop=name,
+                    ref_model=model.name,
+                )
+        else:
+            ref_prop = properties[name]
+            model = models[ref_prop.dtype.model]
+            _check_denorm_prop(name_parts[1], prop, model, manifest)
+    else:
+        if name not in properties:
+            raise UndefinedPropertyInDenormProperty(
+                denorm_prop=prop.name,
+                denorm_model=prop.model.name,
+                ref_prop=name,
+                ref_model=model.name,
+            )
```

Comment:
```suggestion
            raise ReferencedPropertyNotFound(
                prop,
                ref={'property': name, 'model': model.name},
            )
```

Always pass a node as a first argument to exception classes. Also, `ReferencedPropertyNotFound` should be subclass of `PropertyNotFound`, because, basically this is the same thing, but slightly in a different context.

Same thing applies to other exceptions in this function.